packages/wrangler/src/deployment-bundle/node-compat.ts

```diff
@@ -95,9 +79,7 @@ export function getNodeCompatMode({
 	node_compat,
 }: Pick<Config, "compatibility_flags" | "node_compat">) {
 	const nodejsCompat = compatibility_flags.includes("nodejs_compat");
-	const nodejsCompatV2 = compatibility_flags.includes(
-		"experimental:nodejs_compat_v2"
-	);
+	const nodejsCompatV2 = compatibility_flags.includes("nodejs_compat_v2");
```

Comment:
`nodejsCompatV2 === true` implies  `nodejsCompat === true` because of the common prefix - this is probably a bug?

Reply:
A few more nits - not directly related to this PR.

- What about renaming `node_compat` to `nodeCompatLegacy`?
- Could setting legacy use `mode` instead of `node_compat`?
- Any reason we are not using enums for flag names and mode?

Reply:
> `nodejsCompatV2 === true` implies `nodejsCompat === true` because of the common prefix - this is probably a bug?

`compatibility_flags` is an array of strings, so these `include()` calls are correct.
E.g.

```
["nodejs_compat_v2"].includes("nodejs_compat") === false
```

Reply:
> What about renaming node_compat to nodeCompatLegacy?

Sure happy to rename.

Reply:
> Any reason we are not using enums for flag names and mode?

We don't tend to use enums much in this code base. It is nice to have strings when debugging.
Any reason to use enums?

Reply:
> Could setting legacy use mode instead of node_compat?

Not sure what you mean here?

To be clear, what this function does is compute the `mode`, but also return flags telling us if any of these properties were turned on, so we can check the validaty of combinations.

Reply:
> Any reason to use enums?

Avoid repeating a string with the risk of a typo.
enum 
- can be strings `enum Mode { V1 = "v1",  Legacy="legacy"}`
- support autocomplete
- support exhaustive switch (i.e. warn/error when adding a value and not updating a switch)
- are easier to search in the code (arguably)

Reply:
> > Any reason to use enums?
> 
> Avoid repeating a string with the risk of a typo. enum
> 
> * can be strings `enum Mode { V1 = "v1",  Legacy="legacy"}`
> * support autocomplete
> * support exhaustive switch (i.e. warn/error when adding a value and not updating a switch)
> * are easier to search in the code (arguably)

Since the type is a narrow collection of const strings we get all those already: https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBDAnmYcByEAmwBSBlAYQgFswBDGAWS1QF44AiAG2AHMyBjRBuAH0YBuARh78GAgEyi4AOwCuTJgG4AUCo4QZAZ3hkAXOhr4ipCtWxx6zNp26qN2+ACMDGbMZLkqNS4JH3NHTgOVyNCTzMfK0kGAMc4TFD3cNNvC3p5RTigpzJEw2STL3M6RgAzCAgGNWwOJjIoVDK5GQ4YAEtNOFZgNOAACgBKJNwU4ppVFS0Ad3aYDgALOH6evqHBuABvFThd4LItVGt2LgY9Hb3Lp0ayAGtVS44Do+Ezi8vd6+A7h72nw8EUnOHz2Xx+732AIYuUwbxBnxu9wh-1QmSYwPhYKRAF8gA

Reply:
> > Not sure what you mean here?
> 
> `legacy: node_compat === true,` -> `legacy: mode === "legacy",`
> 
> Just to avoid repeating logic as repeating might lead to out of sync code

Ah I see! I can create an extra var to DRY that.